ckan/config/config_declaration.yaml

```diff
@@ -318,7 +318,7 @@ groups:
           The dbm option is not recommended as it is not thread-safe.
 
       - key: beaker.session.validate_key
-        validators: not_empty
+        validators: configured_default("beaker.session.secret",None) not_empty
```

Comment:
Missing space after comma

Reply:
This gave rise to the below error.

```
ckan.logic.UnknownValidator: Validator `configured_default("beaker.session.secret",` does not exist
unable to load app 0 (mountpoint='') (callable not found or import error)
*** no app loaded. going in full dynamic mode ***
Traceback (most recent call last):
  File "/usr/lib/ckan/default/src/ckan/ckan/logic/__init__.py", line 848, in get_validator
    return _validators_cache[validator]
KeyError: 'configured_default("beaker.session.secret",'
```

Reply:
@ThrawnCA , no, that's intended syntax. `validators` evaluated using a strategy from ckanext-scheming: split `v1 v2 v3` on whitespace and then convert every chunk into validator. If you add a space after the comma, you'll get three validators: `configured_default("...",`, `None)` and `not_empty`. And an error, because first two validators do not exist. 

@Suryakandukuri attached example of error that I'm talking about

Reply:
@smotornyuk maybe we could add this explanation about spaces as a note in the docs or in the changelog?

Reply:
It seems rather incongruous for the parser to recognise parenthesised function calls, and yet not be able to apply the parentheses when parsing spaces.

Reply:
Just a bit of explanation, why we are not improving it. The whole "validators" idea for config declaration was taken from ckanext-scheming, because it's arguably most used extension for schema modifications. We don't wan't to use different logic, just to share concepts.
It's possible to write a recursive regular expression using 3rd party library, but I prefer not to add new dependencies, especially for such a small task.
As an alternative, we can make a smarter divider, that counts all the quotes, brackets, parenthesis and splits the validator line only when we have all these symbols in balanced state.

But keep in mind, that our goal it to be scheming-friendly. Any change we do here must be adopted by scheming, so that people remain on the same ground. The question is: does this small inconvenience worth keeping two copies of more complex lexer in two different repos?

If in future scheming integrated into CKAN, it may have sense. Or if we see that people often use validator factories with multiple arguments(I think we have just two such validators in CKAN core and they are not the most popular ones). Otherwise, possibility of adding a space between arguments doesn't worth efforts. At least, I don't have time for making this, but I'll definitely will support a PR with such change